cloudbuild.yaml

```diff
@@ -30,83 +37,133 @@ steps:
+            "site": "'"${_FIREBASE_SITE_ID}"'",
+            "public": "webplugin/build",
+            "ignore": [
+              "firebase.json",
+              "**/.*",
+              "**/node_modules/**"
             ],
-            \"headers\": [
+            "headers": [
               {
-                \"source\": \"**\",
-                \"headers\": [
+                "source": "**",
+                "headers": [
                   {
-                    \"key\": \"Access-Control-Allow-Origin\",
-                    \"value\": \"*\"
+                    "key": "Access-Control-Allow-Origin",
+                    "value": "*"
                   }
                 ]
               },
               {
-                \"source\": \"/v2/kommunicate.app\",
-                \"headers\": [
+                "source": "/v2/kommunicate.app",
+                "headers": [
                   {
-                    \"key\": \"Content-Type\",
-                    \"value\": \"text/javascript\"
+                    "key": "Content-Type",
+                    "value": "text/javascript"
                   },
                   {
-                    \"key\": \"Cache-Control\",
-                    \"value\": \"public, max-age=180\"
+                    "key": "Cache-Control",
+                    "value": "public, max-age=180"
                   }
                 ]
               },
               {
-                \"source\": \"**/*.js\",
-                \"headers\": [
+                "source": "**/*.js",
+                "headers": [
                   {
-                    \"key\": \"Cache-Control\",
-                    \"value\": \"public, max-age=31536000\"
+                    "key": "Cache-Control",
+                    "value": "public, max-age=31536000"
                   }
                 ]
               },
               {
-                \"source\": \"**/*min.css\",
-                \"headers\": [
+                "source": "**/*min.css",
+                "headers": [
                   {
-                    \"key\": \"Cache-Control\",
-                    \"value\": \"public, max-age=31536000\"
+                    "key": "Cache-Control",
+                    "value": "public, max-age=31536000"
                   }
                 ]
               },
               {
-                \"source\": \"resources/**/*.html\",
-                \"headers\": [
+                "source": "resources/**/*.html",
+                "headers": [
                   {
-                    \"key\": \"Cache-Control\",
-                    \"value\": \"public, max-age=31536000\"
+                    "key": "Cache-Control",
+                    "value": "public, max-age=31536000"
                   }
                 ]
               }
+            ],
+            "rewrites": [
+              {
+                "source": "/<*>",
+                "destination": "/index.html",
+                "type": "404"
+              },
+              {

```

Comment:
@GhiyasKM as discussed, simplified the script and use exiting firbebase.json file